### ISSUE TITLE
Trusted Network- Login page 

### ISSUE THREAD

**Issue by tcdrake76** [b0mbays/continuously_casting_dashboards#61]
I can't get this to work as I keep getting the login page. I have reviewed other tickets but didn't come up with a solution. Hopefully I am just missing something obvious. Hopefully someone can help me out here. 

Here is my setup....

Chromecast hooked up to a monitor via HDMI

YAML config:

````
homeassistant:
  external_url: "https://xxxxxx.com" #external URL is accessible from outside of my network.
  auth_providers:
    - type: trusted_networks
      trusted_networks:
        - 172.168.90.35/32 #These are my display IP addresses, replace them with your own (including the /32)
      trusted_users:
        172.168.90.35: 3c9814e2460c4389bbd93c5b98321941 #Ensure this user has admin rights too
      allow_bypass_login: true
    - type: homeassistant


continuously_casting_dashboards:
  logging_level: debug #Required: Set the logging level - debug/info/warning (default is 'warning' - try 'debug' for debugging)
  cast_delay: 60 #Required: Time (in seconds) for casting checks between each device.
  start_time: "01:00"
  end_time: "23:59"
  devices:
    "Living room TV": #Required: Display name of your device. Find this on the actual device's settings or inside the Google Home app.
     - dashboard_url: "https://xxxxxxx.com/lovelace/calendar" #URL is accessible from outside of my network
       volume: 10 #Optional: Volume to set the display. (If you remove this, the device will remain the same volume)

**Comment by b0mbays** [comment]
Hi @tcdrake76 unfortunately I don't have a TV chromecast to test this on but that could the issue. I've only ever tested this on Google Hub's. Did you get this working or still an issue?

**Comment by tcdrake76** [comment]
I am still having the issue. Attached is the login page I get. For some reason I can't get my trusted user (admin) to automatically login. 
![1efe16d7a003fa507ce86ead86b406b78f5db532](https://github.com/b0mbays/continuously_casting_dashboards/assets/125578234/4f394e6a-3cf6-421f-9653-130fe2d496d2)

**Comment by jopper1988** [comment]
Unfortunately i have the same problem with a Google Nest Hub gen2. He won't login into the dashboard:

![20240702_103633 1](https://github.com/b0mbays/continuously_casting_dashboards/assets/5737203/0527f474-b8d4-425f-882a-2c8dd4f0e459)

**Comment by b0mbays** [comment]
Hi @jopper1988 ! Could you replace:

```
trusted_networks:
- 192.168.1.129/24 
```

with:

```
trusted_networks:
- 192.168.1.129/32

```
That should get things going 👍

**Comment by b0mbays** [comment]
I'm assuming that worked @jopper1988? I'll close the issue. I can re-open if not 😄

**Comment by b0mbays** [comment]
Re-opened as realised this was the original issue. Looks like we have an issue with TV chromecast devices of which I can't test as I do not own one. If someone does, and they can implement a fix, please give it a go 👍

**Comment by Mithrox** [comment]
I am getting the same issue as @jopper1988, I am trying to cast to a Nest Hub and I also just get the login screen. When I press in the field for user or password I get the onscreen keyboard, but no buttons get registered when I try to enter user/pass. I have doublechecked all my config, but pasting it here incase someone else see  any glaring mistakes.

`homeassistant:
  external_url: "https://nabu-casa-url/"
  auth_providers:
    - type: trusted_networks
      trusted_networks:
        - 10.10.10.113/32
      trusted_users:
        10.10.10.113: a87a0f6xxxxxxxxxxxxxxxxxxx03f3a8d
      allow_bypass_login: true
    - type: homeassistant

continuously_casting_dashboards:
  logging_level: warning
  cast_delay: 45
  start_time: "07:00"
  end_time: "01:00"
  devices:
    "Loft Display":
      - dashboard_url: "https://nabu-casa-url/lovelace-galaxy/0?kiosk"
        volume: 5
        start_time: "07:00"
        end_time: "01:00"`

**Comment by b0mbays** [comment]
That's strange... Could you double check the IP address for the hub? Is this the same IP that is shown inside the Google Home app when you go to the Hubs settings -> Device information? 10.10.10.113

**Comment by Mithrox** [comment]
> That's strange... Could you double check the IP address for the hub? Is this the same IP that is shown inside the Google Home app when you go to the Hubs settings -> Device information? 10.10.10.113

Yes, it is the correct IP, also read from Unifi manager. I am out right now but will triplecheck when I get back home. 

Bottom line, the login screen should not show when configured correctly?

**Comment by b0mbays** [comment]
Yeah if everything is setup you should not see a login screen at all. You will go straight to the dashboard on the device... Can you ensure the user you have used is an Administrator within home assistant?

**Comment by Mithrox** [comment]
It is the system owner, so default an admin.

**Comment by b0mbays** [comment]
Can you try adding a new user as an admin, and update the configuration to include the ID of that new user?

**Comment by Mithrox** [comment]
> Can you try adding a new user as an admin, and update the configuration to include the ID of that new user?

Unfortunately it was the same result. I created a "cast" user, made it administrator, and then changed the configuration.yaml file and restarted, same result.

I find it weird that touch does not work inside the HA window on the Hub, I would expect to be able to type in user and password, but nothing registers. There is nothing wrong with touch, it works perfectly fine to do native Google Hub actions on the screen.

**Comment by b0mbays** [comment]
> > Can you try adding a new user as an admin, and update the configuration to include the ID of that new user?
> 
> Unfortunately it was the same result. I created a "cast" user, made it administrator, and then changed the configuration.yaml file and restarted, same result.
> 
> I find it weird that touch does not work inside the HA window on the Hub, I would expect to be able to type in user and password, but nothing registers. There is nothing wrong with touch, it works perfectly fine to do native Google Hub actions on the screen.

Could you post your full configuration.yaml file with the formatting? You could upload it as a file or use the code formatting option 👍  Feel free to remove any sensitive IP addresses etc.

Regarding the inability to type, I believe Google deliberately locked the browsers down on these hubs unfortunately.

**Comment by Mithrox** [comment]
[configuration.yaml.txt](https://github.com/user-attachments/files/19167873/configuration.yaml.txt)

**Comment by b0mbays** [comment]
Oh that's great you found it! Nice 😄  If you update a dashboard on another device, and hit save then you should see a popup on the Hubs dashboard in the bottom left with something along the lines of "This dashboard has changed, do you want to refresh?" Then tap that and it will refresh!

**Comment by Mithrox** [comment]
Thank you so much for your support and patience, sorry for the screwup on the code.

**Comment by b0mbays** [comment]
No worries. Enjoy! I'll look to do some more updates on the integration soon, just been busy. But I've been using it now for over 2 years with 5 hubs with minimal issues. I'd recommend setting up a daily restart of your HA instance just to keep things fresh. The networking between the hubs and HA can sometimes be flaky, so be prepared for occasional restarts 👍

**Comment by jopper1988** [comment]
Hello,

Sorry for the late reply, but I’ve been very busy with work. I’ve reviewed the configuration again, but I’m still getting the login screen. For completeness, I’ve attached my configuration.

[# Continuously Casting Dashboards config.txt](https://github.com/user-attachments/files/19177836/Continuously.Casting.Dashboards.config.txt)

**Comment by b0mbays** [comment]
Are you missing your external URL?

Here is my config:

```
homeassistant:
  external_url: "this-is-my-url.naba.casu"
  auth_providers:
    - type: trusted_networks
      trusted_networks:
        - 192.168.0.12/32 #Office display 
        - 192.168.0.9/32 #Emily office display 
        - 192.168.0.17/32 #Living room display
        - 192.168.0.16/32 #Basement display
        - 192.168.0.6/32 #Kitchen display 
      trusted_users:
        192.168.0.12: --user-id #Office display
        192.168.0.9: --user-id  #Emily office display
        192.168.0.17: --user-id  #Living room display
        192.168.0.16: --user-id  #Basement display
        192.168.0.6: --user-id  #Kitchen display
      allow_bypass_login: true
    - type: homeassistant
```

**Comment by jopper1988** [comment]
"I added external_url to the configuration, but it still shows the login screen." Below is my configuration:

`# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

homeassistant:
  external_url: https://stoffelsha.duckdns.org
  packages: !include_dir_named packages
  customize: !include customize.yaml

  # Continuously Casting Dashboards config
auth:
  - providers:
      - type: trusted_networks
        trusted_networks:
          - 192.168.1.129/32 #These are my display IP addresses, replace them with your own (including the /32)
        trusted_users:
          192.168.1.223: eecfd426967d46edb692641173959777 #Ensure this user has admin rights too
        allow_bypass_login: true
      - type: homeassistant

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml
template: !include templates.yaml
sensor: !include sensors.yaml
group: !include groepen.yaml

# Afvalwijzer
input_boolean:
  waste_moved:
    name: Waste has been moved
    initial: off
    icon: mdi:delete-empty
  waste_reminder:
    name: Waste reminder enabled
    initial: on

# Continuously Casting Dashboard
continuously_casting_dashboards:
  logging_level: warning # Required: Set the logging level - debug/info/warning (default is 'warning' - try 'debug' for debugging)
  cast_delay: 45 # Required: Time (in seconds) for casting checks between each device.
  start_time: "07:00" # Optional: Global start time of the casting window (format: "HH:MM") - Default is set to "07:00" and can be individually overwritten per device below.
  end_time: "01:00" # Optional: Global end time of the casting window (format: "HH:MM") and must be after "00:00". Default is set to "01:00" and can be individually overwritten per device below.
  devices:
    "Hub": # Required: Display name of your device. Find this on the actual device's settings or inside the Google Home app.
      - dashboard_url: "http://192.168.1.223:8123/lovelace/home-nieuw?kiosk" # Required: Dashboard URL to be casted (This must be the local IP address of your HA instance, not homeassistant.local)
        volume: 2 # Optional: Volume to set the display. (If you remove this, the device will remain the same volume)
        start_time: "07:00" # Optional: Set the start time for this device
        end_time: "01:00" # Optional: Set the end time for this device
`

**Comment by b0mbays** [comment]
Could you paste that again but use the code formatting instead to retain formatting?  thanks!

**Comment by jopper1988** [comment]
`# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

homeassistant:
  external_url: https://stoffelsha.duckdns.org
  packages: !include_dir_named packages
  customize: !include customize.yaml

  # Continuously Casting Dashboards config
auth:
  providers:
    - type: trusted_networks
      trusted_networks:
        - 192.168.1.129/32 # These are my display IP addresses, replace them with your own (including the /32)
      trusted_users:
        192.168.1.223: eecfd426967d46edb692641173959777 # Ensure this user has admin rights too
      allow_bypass_login: true
    - type: homeassistant

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml
template: !include templates.yaml
sensor: !include sensors.yaml
group: !include groepen.yaml

# Power calculaties
powercalc:

# Whatsapp
whatsapp:

# Afvalwijzer
input_boolean:
  waste_moved:
    name: Waste has been moved
    initial: off
    icon: mdi:delete-empty
  waste_reminder:
    name: Waste reminder enabled
    initial: on

# Continuously Casting Dashboard
continuously_casting_dashboards:
  logging_level: warning # Required: Set the logging level - debug/info/warning (default is 'warning' - try 'debug' for debugging)
  cast_delay: 45 # Required: Time (in seconds) for casting checks between each device.
  start_time: "07:00" # Optional: Global start time of the casting window (format: "HH:MM") - Default is set to "07:00" and can be individually overwritten per device below.
  end_time: "01:00" # Optional: Global end time of the casting window (format: "HH:MM") and must be after "00:00". Default is set to "01:00" and can be individually overwritten per device below.
  devices:
    "Hub": # Required: Display name of your device. Find this on the actual device's settings or inside the Google Home app.
      - dashboard_url: "http://192.168.1.223:8123/lovelace/home-nieuw?kiosk" # Required: Dashboard URL to be casted (This must be the local IP address of your HA instance, not homeassistant.local)
        volume: 2 # Optional: Volume to set the display. (If you remove this, the device will remain the same volume)
        start_time: "07:00" # Optional: Set the start time for this device
        end_time: "01:00" # Optional: Set the end time for this device

# Examples of other devices:
# "Office display":
#   - dashboard_url: "http://192.168.12.104:8123/nest-dashboard/default_view?kiosk"
#     volume: 7
#     start_time: "06:00"
#     end_time: "18:00"
# "Kitchen display":
#   - dashboard_url: "http://192.168.12.104:8123/kitchen-dashboard/default_view?kiosk"
#     volume: 9
#     start_time: "06:00"
#     end_time: "22:00"
# "Basement display":
#   - dashboard_url: "http://192.168.12.104:8123/nest-dashboard/default_view?kiosk"
#     volume: 4
#     start_time: "18:00"
#     end_time: "03:00"
#     speaker_groups:
#       - "Upstairs Speakers"
#       - "Downstairs Speakers"

http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24
  ip_ban_enabled: false # Enable auto IP ban
  login_attempts_threshold: 3 # Allowed number of login attempts
`

**Comment by b0mbays** [comment]
Your configuration looks incorrect, please replace your homeassistant section with this:

```
homeassistant:
  external_url: https://stoffelsha.duckdns.org
  packages: !include_dir_named packages
  customize: !include customize.yaml
  auth_providers:
      - type: trusted_networks
        trusted_networks:
          - 192.168.1.129/32 # These are my display IP addresses, replace them with your own (including the /32)
        trusted_users:
          192.168.1.223: eecfd426967d46edb692641173959777 # Ensure this user has admin rights too
        allow_bypass_login: true
      - type: homeassistant

```

**Comment by jopper1988** [comment]
Tnx for your help it now works :-)

**Comment by b0mbays** [comment]
Great to hear 😄
